router/link/link_registry.go

```diff
@@ -325,13 +325,31 @@ func (self *linkRegistryImpl) evaluateLinkStateQueue() {
 }
 
 func (self *linkRegistryImpl) evaluateDestinations() {
-	for _, dest := range self.destinations {
-		// TODO: When do we drop destinations? Should we ask the controller after the router has been
-		//       unhealthy for a while and it doesn't have any established links? Do this on exponential backoff?
-		//      Should the controller send router removed messages?
+	for destId, dest := range self.destinations {
+		hasEstablishedLinks := false
 		for _, state := range dest.linkMap {
+			// verify that links marked as established have an open link. There's a small chance that a link established
+			// and link closed could be processed out of order if the event queue is full. This way, it will eventually
+			// get fixed.
+			if state.status == StatusEstablished {
+				link, _ := self.GetLink(state.linkKey)
+				if link == nil || link.IsClosed() {
+					state.retryDelay = time.Duration(0)
```

Comment:
it might be helpful to add why the retryDelay would be set to 0 and what might unset it from 0 in the future (if needed)

Reply:
Hmm, this should be resetting the link status as well.

Reply:
Fixed and added a comment